test/unit/utils.test.ts

```diff
@@ -1101,4 +1104,65 @@ describe('driver utils', function () {
       });
     });
   });
+
+  describe('class TimeoutController', () => {
+    let timerSandbox, clock, spy;
+
+    beforeEach(function () {
+      timerSandbox = createTimerSandbox();
+      clock = sinon.useFakeTimers();
+      spy = sinon.spy();
+    });
+
+    afterEach(function () {
+      clock.restore();
+      timerSandbox.restore();
+    });
+
+    describe('constructor', () => {
+      it('when no timeout is provided, it creates an infinite timeout', () => {
+        const controller = new TimeoutController();
+        // @ts-expect-error Accessing a private field on TimeoutController
+        expect(controller.timeoutId).to.be.null;
+      });
+
+      it('when timeout is 0, it creates an infinite timeout', () => {
+        const controller = new TimeoutController(0);
+        // @ts-expect-error Accessing a private field on TimeoutController
+        expect(controller.timeoutId).to.be.null;
+      });
+
+      it('when timeout <0, it creates an infinite timeout', () => {
+        const controller = new TimeoutController(-5);
+        // @ts-expect-error Accessing a private field on TimeoutController
+        expect(controller.timeoutId).to.be.null;
+      });
+
+      context('when timeout > 0', () => {
+        let timeoutController: TimeoutController;
+
+        beforeEach(function () {
+          timeoutController = new TimeoutController(3000);
+          timeoutController.signal.onabort = spy;
+        });
+
+        afterEach(function () {
+          timeoutController.clear();
+        });
+
+        it('it creates a timeout', () => {
+          // @ts-expect-error Accessing a private field on TimeoutController
+          expect(timeoutController.timeoutId).not.to.be.null;
+        });
+
+        it('times out after `timeout` milliseconds', () => {
+          expect(spy.called, 'spy was called after creation').to.be.false;
```

Comment:
we can enhance this with the chai sinon assertions: to.be.called right?

Reply:
Would you prefer that I use the chai assertions?

Reply:
sí, por favor